### ISSUE TITLE
Cannot bundle update JRuby SQL Server

### ISSUE THREAD

**Issue by chemturion** [jruby/activerecord-jdbc-adapter#1046]
I'm trying to upgrade JRuby on Rails 4 to 5, but I can't get the SQL Server adapter to install, so I'm not sure if I'm specifying things as they are intended. I cannot specify the branch on GitHub in the Gemfile due to SSL cert issue, so I downloaded the zip from the [50-stable branch repo](https://github.com/jruby/activerecord-jdbc-adapter/tree/50-stable). This part of my Gemfile is as follows:

### Gemfile
```ruby
source 'http://rubygems.org'

gem 'rails', '5.0.0'
gem 'activerecord-jdbcsqlite3-adapter' # I think I still need this one for local tests
gem 'activerecord'
gem 'activerecord-jdbc-adapter', path: 'C:/projects/activerecord-jdbc-adapter-50-stable'
```
* `bundle install` produces this *compatible versions* warning:
```
Bundler could not find compatible versions for gem "activerecord":
  In snapshot (Gemfile.lock):
    activerecord (= 5.0.0)

  In Gemfile:
    activerecord-jdbc-adapter java was resolved to 50.5, which depends on
      activerecord (~> 5.0.0, >= 5.0.3) java

    annotate java was resolved to 3.0.3, which depends on
      activerecord (>= 3.2, < 7.0) java

    kaminari java was resolved to 1.1.1, which depends on
kaminari-activerecord (= 1.1.1) java was resolved to 1.1.1, which depends
on
        activerecord java

    paper_trail java was resolved to 10.3.1, which depends on
      activerecord (>= 4.2) java

    rails (= 5.0.0) java was resolved to 5.0.0, which depends on
      activerecord (= 5.0.0) java

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```
* And `bundle update` produces a similar warning:
```
Bundler could not find compatible versions for gem "activerecord":
  In Gemfile:
    activerecord java

    activerecord-jdbc-adapter java was resolved to 50.5, which depends on
      activerecord (~> 5.0.0, >= 5.0.3) java

    rails (= 5.0.0) java was resolved to 5.0.0, which depends on
      activerecord (= 5.0.0) java
```

### Alternative Gemfile
I thought perhaps `activerecord-jdbc-adapter` should be `activerecord-jdbcsqlserver-adapter`, but doing that produces a different warning after attempting `bundle install`:
```ruby
# gem 'activerecord-jdbc-adapter', path: 'C:/projects/activerecord-jdbc-adapter-50-stable'
gem 'activerecord-jdbcsqlserver-adapter', path: 'C:/projects/activerecord-jdbc-adapter-50-stable'
```
```
Could not find gem 'activerecord-jdbcsqlserver-adapter' in source at
`C:/projects/activerecord-jdbc-adapter-50-stable` or in gems cached in
vendor/cache.
The source does not contain any versions of 'activerecord-jdbcsqlserver-adapter'
```

I think I've configured things as y'all intend with this JRuby <> SQL Server Adapter combination, but I just want to check to rule out any misunderstanding on my part. Thank you for hearing me out! 🙏

**Comment by rdubya** [comment]
Hi @chemturion,
Have you tried including `gem 'activerecord-jdbcsqlserver-adapter', '~>50.0'` in your gemfile? That is the released version that should support the version of rails you are trying to upgrade to.

**Comment by chemturion** [comment]
Let me try that instead. I also just realized that the Gemfile.lock has `activerecord (5.0.0)`, but the error has both squiggly and greater-than constraint (no more than 5.x, and greater-than 5.0.3), so I think I need to bump my Rails version to 5.0.3 because 5.0.0 may be the culprit right now, no? 🤔 

```
In Gemfile:
    activerecord-jdbc-adapter java was resolved to 50.5, which depends on
      activerecord (~> 5.0.0, >= 5.0.3) java
```

**Comment by rdubya** [comment]
Ah yeah, good catch. You may additionally need to specify the version for the activerecord-jdbcsqlserver-adapter gem as well, not sure though.

**Comment by chemturion** [comment]
Awesome, thanks for your help. I updated my Gemfile to use the `activerecord-jdbcsqlserver-adapter` and removed the local "path" one I referenced earlier. I got a slightly different dependency warning than last time so I bumped my Rails version to 5.0.6. For posterity, here's that warning and my updated Gemfile that resolved it:

```
Bundler could not find compatible versions for gem "activerecord":
  In snapshot (Gemfile.lock):
    activerecord (= 5.0.3)

  In Gemfile:
activerecord-jdbcsqlserver-adapter (~> 50.0) java was resolved to 50.0.0,
which depends on
      activerecord (~> 5.0.0, >= 5.0.6) java

    rails (= 5.0.3) java was resolved to 5.0.3, which depends on
      activerecord (= 5.0.3) java

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

### Gemfile
```ruby
gem 'rails', '5.0.6'
gem 'activerecord-jdbcsqlite3-adapter'
gem 'activerecord'
gem 'activerecord-jdbcsqlserver-adapter', '~>50.0'
```
